static/src/javascripts/projects/common/modules/discussion/loader.js

```diff
@@ -0,0 +1,580 @@
+// @flow
+
+import bean from 'bean';
+import bonzo from 'bonzo';
+import qwery from 'qwery';
+import $ from 'lib/$';
+import raven from 'lib/raven';
+import config from 'lib/config';
+import { isBreakpoint } from 'lib/detect';
+import mediator from 'lib/mediator';
+import { scrollToElement } from 'lib/scroller';
+import fastdom from 'lib/fastdom-promise';
+import fetchJson from 'lib/fetch-json';
+import DiscussionAnalytics from 'common/modules/analytics/discussion';
+import register from 'common/modules/analytics/register';
+import Component from 'common/modules/component';
+import DiscussionApi from 'common/modules/discussion/api';
+import { CommentBox } from 'common/modules/discussion/comment-box';
+import Comments from 'common/modules/discussion/comments';
+import discussionFrontend from 'common/modules/discussion/discussion-frontend';
+import {
+    handle as upvoteHandle,
+    closeTooltip as upvoteCloseTooltip,
+} from 'common/modules/discussion/upvote';
+import Id from 'common/modules/identity/api';
+import userPrefs from 'common/modules/user-prefs';
+
+class Loader extends Component {
+    constructor(): void {
+        super();
+
+        this.classes = {};
+        this.componentClass = 'discussion';
+        this.comments = null;
+        this.user = null;
+        this.username = null;
+
+        register.begin('discussion');
+    }
+
+    getUser(): void {
+        if (Id.getUserFromCookie()) {
+            DiscussionApi.getUser().then(resp => {
+                this.user = resp.userProfile;
+
+                Id.getUserFromApi(user => {
+                    if (user && user.publicFields.username) {
+                        this.username = user.publicFields.username;
+                    }
+
+                    this.emit('user:loaded');
+                });
+            });
+        } else {
+            this.emit('user:loaded');
+        }
+    }
+
+    getDiscussionId(): ?string {
+        return this.elem.getAttribute('data-discussion-key');
+    }
+
+    getDiscussionClosed(): boolean {
+        return this.elem.getAttribute('data-discussion-closed') === 'true';
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    getCommentIdFromHash(): ?number {
+        const reg = /#comment-(\d+)/;
+        return reg.exec(window.location.hash)
+            ? parseInt(reg.exec(window.location.hash)[1], 10)
+            : null;
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    setCommentHash(id: string): void {
+        window.location.replace(`#comment-${id}`);
+    }
+
+    commentPosted(...args: Array<mixed>): void {
```

Comment:
A full text search on `'post:success'` reveals that this method only ever receives one argument (`comment: commentType`)